segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/FixedKeyLengthTableSegmentLayout.java

```diff
@@ -125,7 +125,7 @@ public void close() {
         ensureValidKeyLength("", config.getKeyLength());
         val result = new HashMap<AttributeId, Long>();
         result.put(Attributes.ATTRIBUTE_ID_LENGTH, (long) config.getKeyLength());
-        result.putAll(this.config.getDefaultCompactionAttributes());
+        result.putAll(this.config.getOrDefaultCompactionAttributes(config));
```

Comment:
I am not sure how I feel about this pattern. The point of the `getDefaultCompactionAttributes` was to fill in default values which would later be overridden. I would prefer if we kept the old mechanism, and then we populate specific attributes (essentially override) based on `config`.

When doing so, remember to treat `0L` as "default" - so skip that value.